### PR TITLE
#75: Consets object null using nodejs (closes #75)

### LINKED ISSUE #75: Consets object null using nodejs
we are fancing same issue on adding information in the proprety consents using the addcustomer/Patchcustomer function, below the payload sent.

It works properly using API service but using nodejs the property consents is null:

{
               "nodeId": "xxx",
               "enabled": true,
               "base": {
                              "firstName": "name",
                              "lastName": "surname",
                              "contacts": {
                                            "email": "test@domain.com",
                                            "phone": "+393471234567"
                              },
                              "address": {
                                            "country": "IT"
                              }
               },
               "extended": {
                              "jobTitle": "Owner / General Manager / CEO",
                              "department": "Business Intelligence",
                              "company": "xxxxxx",
                              "sector": "OTR - System Integrator",
                              "preferences": {
                                            "read_privacy": "true",
                                            "promotional": "false",
                                            "profiling": "false"
                              },
                              "touchpoint": "webinar",
                              "lang": "it",
                              "emailVerified": false
               },
               "tags": {
               },
               "consents": {
                              "disclaimer": {
                                            "date": "2018-05-25T00:00:00.000Z",
                                            "version": "1-2018"
                              },
                              "marketing": {
                                            "traditional": {
                                                           "telephonic": {
                                                                          "status": false
                                                           },
                                                           "papery": {
                                                                          "status": false
                                                           }
                                            },
                                            "automatic": {
                                                           "sms": {
                                                                          "status": false
                                                           },
                                                           "email": {
                                                                          "status": true
                                                           },
                                                           "push": {
                                                                          "status": false
                                                           },
                                                           "im": {
                                                                          "status": false
                                                           }
                                            }
                              },
                              "profiling": {
                                            "classic": {
                                                           "status": false
                                            },
                                            "online": {
                                                           "status": false
                                            }
                              }
               }
}

### PR DESCRIPTION
**Pull request by dhinus**
Closes #75

### tests performed

* Added new e2e and unit tests
* Tested manually using the Node.js REPL:

```
> ch.addCustomer(customerData).then(data => { console.log(data); });
Promise {
  <pending>,
  domain:
   Domain {
     domain: null,
     _events:
      { removeListener: [Function: updateExceptionCapture],
        newListener: [Function: updateExceptionCapture],
        error: [Function: debugDomainError] },
     _eventsCount: 3,
     _maxListeners: undefined,
     members: [] } }
> { id: '26bc3cbe-2929-40a7-aa6b-3c4f595fa66c',
  registeredAt: 2018-07-03T14:54:33.993Z,
  updatedAt: 2018-07-03T14:54:33.993Z,
  base:
   { firstName: 'Test',
     lastName: 'Manuale',
     contacts: { email: 'qwiudhqwiudhwqiu@example.com' } },
  consents:
   { disclaimer: { date: 2018-05-25T00:00:00.000Z, version: 'v1' },
     marketing: { traditional: [Object] } } }
```

* Verified the Consents are visible in the ContactHub web interface:

screen shot 2018-07-03 at 16 55 14

### CODE REVIEW

File: src/ContactHub.js [hunk 1]
```diff
@@ -85,17 +75,24 @@ const cleanCustomer = (data: APICustomer): Customer => {
...
+    // check for data.base is useful only to avoid a Flow warning
     if (data.base && data.base.dob) {
       customer.base.dob = new Date(data.base.dob);
     }
 
   }
 
+  if (data.consents) {
+    customer.consents = (compact(data.consents));
+    if (data.consents && data.consents.disclaimer && data.consents.disclaimer.date) {
```
Comment (veej) [inline src/ContactHub.js]: minor: you're already checking if `data.consents` is null in the external if.

File: src/ContactHub.js [hunk 2]
```diff
@@ -85,17 +75,24 @@ const cleanCustomer = (data: APICustomer): Customer => {
...
+      customer.consents.disclaimer.date = new Date(data.consents.disclaimer.date);
```
Comment (veej) [inline src/ContactHub.js]: you are no longer using `formatToDate`. Is it right?

**Review by veej — APPROVED** [review]
two minor comments. Feel free to ignore them

File: src/ContactHub.js (hunk 1 above)
Comment (dhinus) [reply src/ContactHub.js]: Flow is complaining if I remove it, because it doesn't trust `compact(data.consents)`.

File: src/ContactHub.js (hunk 2 above)
Comment (dhinus) [reply src/ContactHub.js]: Yes, that was one of the issues reported via email (not in the GitHub issue): the field is a datetime in the API, and using `formatToDate` was cutting the time off.

**Pull request merged**